internal/lsp/server.go

```diff
@@ -790,6 +792,14 @@ func (l *LanguageServer) templateContentsForFile(fileURI string) (string, error)
 		return "", errors.New("file already has contents, templating not allowed")
 	}
 
+	diskContent, err := os.ReadFile(uri.ToPath(l.clientIdentifier, fileURI))
```

Comment:
this is also not the fix but is an additional guard for templating over files that have contents on disk already.

Reply:
Any added safeguards are good 👍